anonymizer/anonymizer.c

```diff
@@ -257,46 +375,61 @@ int main(int argc, char **argv)
          break;
       default:
          fprintf(stderr, "Invalid arguments.\n");
-         FREE_MODULE_INFO_STRUCT(MODULE_BASIC_INFO, MODULE_PARAMS)
-         return 3;
+         ret = 1;
+         goto cleanup;
       }
    }
 
    // Check if secret key was specified and initialize panonymizer
    if (secret_file != NULL) {
       if (!init_from_file(secret_file, init_key)) {
-         trap_finalize();
-         FREE_MODULE_INFO_STRUCT(MODULE_BASIC_INFO, MODULE_PARAMS)
-         return 7;
+         ret = 2;
+         goto cleanup;
       }
    }
    else {
       if (!ParseCryptoPAnKey(secret_key, init_key)) {
-         trap_finalize();
-         FREE_MODULE_INFO_STRUCT(MODULE_BASIC_INFO, MODULE_PARAMS)
-         return 7;
+         ret = 3;
+         goto cleanup;
       }
       PAnonymizer_Init(init_key);
    }
-
    // ***** Create UniRec input template *****
-   char *unirec_specifier = "SRC_IP,DST_IP";
-   char *errstr = NULL;
-   ur_template_t *tmplt = ur_create_input_template(0, unirec_specifier, &errstr);
-   trap_set_required_fmt(0, TRAP_FMT_UNIREC, "ipaddr SRC_IP,ipaddr DST_IP");
 
+   tmplt = ur_create_input_template(0, NULL, NULL);
+   trap_set_required_fmt(0, TRAP_FMT_UNIREC, NULL);
    if (tmplt == NULL) {
-      fprintf(stderr, "Error: Invalid UniRec specifier.\n");
-      if(errstr != NULL){
-        fprintf(stderr, "%s\n", errstr);
-        free(errstr);
-      }
-      trap_finalize();
-      FREE_MODULE_INFO_STRUCT(MODULE_BASIC_INFO, MODULE_PARAMS)
-      return 4;
+      fprintf(stderr, "Error: Unable to create input template.\n");
+         ret = 4;
+         goto cleanup;
+   }
+
+   anon_rec = calloc(UR_MAX_SIZE, 1);
+   if (!anon_rec) {
+      fprintf(stderr, "Error: Memory allocation problem (output alert record).\n");
+         ret = 5;
+         goto cleanup;
+   }
+
+   reti = regcomp(&regex_IPV4, "[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}", REG_EXTENDED);
+   if (reti) {
+         ret = 6;
+         goto cleanup;
+   }
+
+   reti = regcomp(&regex_IPV6, "(([0-9a-fA-F]{1,4}:) {7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:) {1,7}:|\
```

Comment:
This const string with regex should be declared as a macro on the top or in the header file...
Same for ipv4.